src/property-filter/i18n-utils.ts

```diff
@@ -1,39 +1,235 @@
+            group__formattedTokens0__formattedText: tokens[0] ? formatToken(tokens[0]) : '',
+            group__formattedTokens1__formattedText: tokens[1] ? formatToken(tokens[0]) : '',
+            group__formattedTokens2__formattedText: tokens[2] ? formatToken(tokens[0]) : '',
+            group__formattedTokens3__formattedText: tokens[3] ? formatToken(tokens[0]) : '',
+          })
+      );
+      return formatter?.({ operation, operationLabel, tokens }) ?? '';
+    },
+    removeTokenButtonAriaLabel: token => {
+      const formatter = i18n(
+        'i18nStrings.removeTokenButtonAriaLabel',
+        def.removeTokenButtonAriaLabel,
+        format => () => format(toI18n(token, { ariaLabel: true }))
+      );
+      return formatter?.(toFormatted(token)) ?? '';
+    },
+    tokenEditorTokenActionsAriaLabel: token => {
+      const formatter = i18n(
+        'i18nStrings.tokenEditorTokenActionsAriaLabel',
+        def.tokenEditorTokenActionsAriaLabel,
+        format => () => format(toI18n(token, { ariaLabel: true }))
+      );
+      return formatter?.(toFormatted(token)) ?? '';
+    },
+    tokenEditorTokenRemoveAriaLabel: token => {
+      const formatter = i18n(
+        'i18nStrings.tokenEditorTokenRemoveAriaLabel',
+        def.tokenEditorTokenRemoveAriaLabel,
+        format => () => format(toI18n(token, { ariaLabel: true }))
+      );
+      return formatter?.(toFormatted(token)) ?? '';
+    },
+    tokenEditorAddExistingTokenAriaLabel: token => {
+      const formatter = i18n(
+        'i18nStrings.tokenEditorAddExistingTokenAriaLabel',
+        def.tokenEditorAddExistingTokenAriaLabel,
+        format => () => format(toI18n(token, { ariaLabel: true }))
+      );
+      return formatter?.(toFormatted(token)) ?? '';
+    },
+    tokenEditorAddExistingTokenLabel: token => {
+      const formatter = i18n(
+        'i18nStrings.tokenEditorAddExistingTokenLabel',
+        def.tokenEditorAddExistingTokenLabel,
+        format => () => format(toI18n(token, { ariaLabel: false }))
+      );
+      return formatter?.(toFormatted(token)) ?? '';
+    },
   };
 }
 
-export function getFormattedToken(
-  token: InternalToken,
-  i18nStrings: I18nStrings
-): FormattedToken & { formattedTokenText: string } {
-  const valueFormatter = token.property?.getValueFormatter(token.operator);
-  const propertyLabel = token.property ? token.property.propertyLabel : i18nStrings.allPropertiesLabel ?? '';
-  const tokenValue = valueFormatter ? valueFormatter(token.value) : token.value;
-  const formatted = {
-    propertyKey: token.property?.propertyKey,
-    propertyLabel,
-    operator: token.operator,
-    value: tokenValue,
-  };
-  const getText = i18nStrings.formatToken ?? (token => `${token.propertyLabel} ${token.operator} ${token.value}`);
-  return { ...formatted, formattedTokenText: getText(formatted) };
+export function operatorToDescription(operator: ComparisonOperator, i18nStrings: I18nStringsInternal) {
```

Comment:
The operatorToDescription function was copied from controller.ts - it is unchanged.